File: fridaClient.js
const axios = require('axios');

function getAgents() {
  const raw = process.env.FRIDA_AGENTS || '{}';
  try {
    const parsed = JSON.parse(raw);
    if (!parsed || typeof parsed !== 'object' || Array.isArray(parsed)) return {};
    return parsed;
  } catch (err) {
    console.error('❌ FRIDA_AGENTS JSON error:', err.message);
    console.error('FRIDA_AGENTS value must be like: {"phone1":"http://1.2.3.4:5000"}');
    return {};
  }
}

async function runFridaScript(deviceId, userId) {
  const agents = getAgents();
  const agentUrl = agents[deviceId];

  if (!agentUrl) return `❌ الجهاز '${deviceId}' غير مسجل في FRIDA_AGENTS`;

  try {
    const response = await axios.post(`${String(agentUrl).replace(/\/$/, '')}/run_frida`, {
      script: 'hook_1m.js',
      user_id: userId
    }, { timeout: 60000 });

    const data = typeof response.data === 'string'
      ? response.data
      : JSON.stringify(response.data, null, 2);

    return `✅ تم التنفيذ على ${deviceId}:\n${data.slice(0, 900)}`;
  } catch (error) {
    if (error.response) {
      return `🔌 فشل الاتصال: HTTP ${error.response.status}\n${JSON.stringify(error.response.data).slice(0, 500)}`;
    }
    return `🔌 فشل الاتصال: ${error.message}`;
  }
}

module.exports = { getAgents, runFridaScript };

File: frida_client.py
import aiohttp
import asyncio
from config import FRIDA_AGENTS

async def run_frida_script(device_id: str, user_id: int, script_path: str = "hook_1m.js") -> str:
    agent_url = FRIDA_AGENTS.get(device_id)
    if not agent_url:
        return f"❌ الجهاز '{device_id}' غير مسجل في FRIDA_AGENTS"

    try:
        timeout = aiohttp.ClientTimeout(total=60)
        async with aiohttp.ClientSession(timeout=timeout) as session:
            payload = {"script": script_path, "user_id": user_id}
            async with session.post(f"{agent_url}/run_frida", json=payload) as resp:
                if resp.status == 200:
                    result_text = await resp.text()
                    return f"✅ تم التنفيذ على {device_id}:\n{result_text[:500]}"
                else:
                    return f"⚠️ خطأ {resp.status} من الجهاز: {await resp.text()}"
    except asyncio.TimeoutError:
        return "⏰ انتهى الوقت – الجهاز لم يستجب خلال 60 ثانية"
    except Exception as e:
        return f"🔌 فشل الاتصال: {str(e)}"